core/src/main/java/io/parsingdata/metal/data/DataExpressionSource.java

```diff
@@ -76,7 +75,7 @@ protected boolean isAvailable(final BigInteger offset, final BigInteger length)
                 .map(Value::getValue)
                 .orElseThrow(() -> new IllegalStateException("ValueExpression dataExpression yields empty Value at index " + index + "."));
         }
-        return cache.clone();
+        return cache;
```

Comment:
This is not needed because of the `arraycopy` above?

Reply:
Exactly. The only client of this method that passes data out does the copying itself.